### PR TITLE
[ENH] KASBA clusterer

### PR DESCRIPTION
**Pull request by chrisholder**
#### What does this implement/fix? Explain your changes.
This PR adds KASBA to aeon. KASBA is a fast and accurate clusterer we developed. More details can be found here: https://arxiv.org/abs/2411.17838

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you.
- [ ] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

##### For new estimators and functions
- [ ] I've added the estimator/function to the online [API documentation](https://www.aeon-toolkit.org/en/latest/api_reference.html).
- [ ] (OPTIONAL) I've added myself as a `__maintainer__` at the top of relevant files and want to be contacted regarding its maintenance. Unmaintained files may be removed. This is for the full file, and you should not add yourself if you are just making minor changes or do not want to help maintain its contents.

##### For developers with write access
- [ ] (OPTIONAL) I've updated aeon's [CODEOWNERS](https://github.com/aeon-toolkit/aeon/blob/main/CODEOWNERS) to receive notifications about future changes to these files.

### CODE REVIEW

File: aeon/clustering/_kasba.py [hunk 1]
```diff
@@ -0,0 +1,363 @@
+"""Time series kmeans."""
+
+from typing import Optional
+
+__maintainer__ = []
```
Comment (TonyBagnall) [inline aeon/clustering/_kasba.py:5]: that would be you :)

File: aeon/clustering/_kasba.py [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+from aeon.clustering.base import BaseClusterer
+from aeon.distances import distance as distance_func
+from aeon.distances import pairwise_distance
+
+
+class KASBA(BaseClusterer):
+    """KASBA clusterer [1]_.
+
+    KASBA is a $k$-means clustering algorithm designed for use with the MSM distance
+    metric [2]_ however, can be used with any elastic distance that is a metric.
```
Comment (TonyBagnall) [inline aeon/clustering/_kasba.py:24]: it can be used

**Review by TonyBagnall — DISMISSED** [review]
LGTM

**Comment by MatthewMiddlehurst** [comment]
Not merging, see above comments

**Review by MatthewMiddlehurst — APPROVED** [review]

**Pull request merged**